components/heap/multi_heap_poisoning.c

```diff
@@ -67,6 +59,9 @@ static uint8_t *poison_allocated_region(poison_head_t *head, size_t alloc_size)
     poison_tail_t *tail = (poison_tail_t *)(data + alloc_size);
     head->alloc_size = alloc_size;
     head->head_canary = HEAD_CANARY_PATTERN;
+#ifdef CONFIG_HEAP_TASK_TRACKING
+    head->task = xTaskGetCurrentTaskHandle();
```

Comment:
One of the (probably unwritten, sorry) design decisions for multi_heap is that it doesn't contain any platform-specific code. This allows us to run, for example, the host-side tests in test_multi_heap_host. Eventually we may split multi_heap out to a standalone project outside IDF.

Would you be able to please move the platform-specific parts of this (config macro and FreeRTOS API call) to multi_heap_config.h and multi_heap_platform.h, respectively? Also, please check the host-side tests still compile & pass.